File: app/src/main/java/com/android/example/popularmovies/utils/json/MovieReview.java
package com.android.example.popularmovies.utils.json;

import android.net.Uri;

import java.util.Objects;

public class MovieReview {

    private String author;
    private String content;
    private String id;
    private String url;

    public MovieReview(String author, String content, String id, String url) {
        this.author = author;
        this.content = content;
        this.id = id;
        this.url = url;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }

    public String getContent() {
        return content;
    }



    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUrl() {
        return url;
    }

    public Uri getUrlAsUri(){
        return Uri.parse(url);
    }

    public void setUrl(String url) {
        this.url = url;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        MovieReview that = (MovieReview) o;
        return author.equals(that.author) &&
                content.equals(that.content) &&
                id.equals(that.id) &&
                url.equals(that.url);
    }

    @Override
    public int hashCode() {
        return Objects.hash(author, content, id, url);
    }
}

File: app/src/main/java/com/android/example/popularmovies/utils/json/MovieReview.kt
package com.android.example.popularmovies.utils.json

import android.net.Uri
import androidx.core.net.toUri
import java.util.*

class MovieReview(val author: String, val content: String, val id: String, val url: String) {

    val urlAsUri: Uri
        get() = url.toUri()

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as MovieReview
        return author == that.author && content == that.content && id == that.id && url == that.url
    }

    override fun hashCode(): Int {
        return Objects.hash(author, content, id, url)
    }

}